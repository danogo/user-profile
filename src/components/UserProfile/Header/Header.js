import React from 'react';
import styles from './Header.module.css';
import picture from './pic.jpg';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Header = (props) => {
  const { onHeartClick, onFollowClick, likes, following, followers, isFollowed, isLiked, onShareClick } = props;
  return (
      <div className={styles.Header}>
        <FontAwesomeIcon onClick={onShareClick} className={styles.shareIcon} icon='share-square' tabIndex='0'/>
        <div className={styles.userInfo}>
          <img src={picture} alt='user'/>
          <div>
            <h1>Harvey Specter</h1>
            {
              isLiked
              ? <FontAwesomeIcon onClick={onHeartClick} className={styles.heartIcon} icon={['fas', 'heart']} color='#FFA640'  tabIndex='0'/>
              : <FontAwesomeIcon onClick={onHeartClick} className={styles.heartIcon} icon={['far', 'heart']} color='#D9D9D9'  tabIndex='0'/>
            }
            <h2>New York, USA</h2>
          </div>
        </div>
        <div className={styles.userSocials}>
          <ul id="ul">
            <li>
              <p className={styles.socialsNumber}>{likes}</p>
              <p>Likes</p>
            </li>
            <span className={styles.divider}></span>
            <li>
              <p className={styles.socialsNumber}>{following}</p>
              <p>Following</p>
            </li>
            <span className={styles.divider}></span>
            <li>
              <p className={styles.socialsNumber}>{followers}</p>
              <p>Followers</p>
            </li>
          </ul>
          {
            isFollowed 
              ? <button onClick={onFollowClick} className={`${styles.btnFollow} ${styles.btnUnfollow}`}>UNFOLLOW</button>
              : <button onClick={onFollowClick} className={styles.btnFollow}>FOLLOW</button>
          }
        </div>
      </div>
    )
}

export default Header;