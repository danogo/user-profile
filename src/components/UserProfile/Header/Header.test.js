import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {
  let wrapper;
  let mockProps;
  beforeEach(() => {
    mockProps = {
      onHeartClick: jest.fn(),
      onFollowClick: jest.fn(),
      onShareClick: jest.fn(),
      likes: 121,
      following: 723,
      followers: 4433,
      isLiked: false,
      isFollowed: false,
    }
    wrapper = shallow(<Header {...mockProps}/>);
  });

  test('component renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('renders div container', () => {
    const divs = wrapper.find('div');
    expect(divs.length).toBeGreaterThan(0);
  });

  test('contains all children components', () => {
    const div = wrapper.find('div').first();
    expect(div.children()).toEqual(wrapper.children());
    expect(wrapper.children().length).toBe(3);
  });

  test('receives all onClick handlers', () => {
    expect(wrapper.find('[id="btnShare"]').props().onClick).toBe(mockProps.onShareClick);
    expect(wrapper.find('[id="btnHeart"]').props().onClick).toBe(mockProps.onHeartClick);
    expect(wrapper.find('[id="btnFollow"]').props().onClick).toBe(mockProps.onFollowClick);
  });

  test('displays socials correctly', () => {
    expect(wrapper.find('.socialsNumber').length).toBe(3);
    expect(wrapper.find('.socialsNumber').at(0).text()).toBe(`${mockProps.likes}`);
    expect(wrapper.find('.socialsNumber').at(1).text()).toBe(`${mockProps.following}`);
    expect(wrapper.find('.socialsNumber').at(2).text()).toBe(`${mockProps.followers}`);
  });

  describe('when isLiked prop is false', () => {
    test('displays regular icon with white color', () => {
      expect(wrapper.find('.heartIcon').props().icon).toEqual(['far', 'heart']);
      expect(wrapper.find('.heartIcon').props().color).toEqual('#D9D9D9');
    });

  });

  describe('when isLiked prop is true', () => {
    beforeEach(() => {
      mockProps.isLiked = true;
      wrapper = shallow(<Header {...mockProps}/>);
    });
    test('display solid icon with orange color', () => {
      expect(wrapper.find('.heartIcon').props().icon).toEqual(['fas', 'heart']);
      expect(wrapper.find('.heartIcon').props().color).toEqual('#FFA640');
    });
  }); 

  describe('when isFollowed prop is false', () => {
    test('button has "FOLLOW" text and only btnFollow class', () => {
      // console.log(wrapper.find('#btnFollow').text())
      // console.log(wrapper.find('#btnFollow').props().className)
      expect(wrapper.find('#btnFollow').text()).toBe('FOLLOW');
      expect(wrapper.find('#btnFollow').props().className).toBe('btnFollow');
    });
  });
  
  describe('when isFollowed prop is true', () => {
    beforeEach(() => {
      mockProps.isFollowed = true;
      wrapper = shallow(<Header {...mockProps}/>);
    });
    test('button has "UNFOLLOW" text and 2 classes: btnFollow, btnUnfollow', () => {
      // console.log(wrapper.find('#btnUnfollow').text())
      // console.log(wrapper.find('#btnUnfollow').props().className)
      expect(wrapper.find('#btnUnfollow').text()).toBe('UNFOLLOW');
      expect(wrapper.find('#btnUnfollow').props().className).toBe('btnFollow btnUnfollow');
    });
  });

  test('callback is called when user clicks btnHeart', () => {
    const btnHeart = wrapper.find('#btnHeart');
    // btnHeart.simulate('click');
    btnHeart.prop('onClick')();
    expect(mockProps.onHeartClick).toHaveBeenCalledTimes(1);
  })

  test('callback is called when user clicks btnShare', () => {
    const btnShare = wrapper.find('#btnShare');
    // btnShare.simulate('click');
    btnShare.prop('onClick')();
    expect(mockProps.onShareClick).toHaveBeenCalledTimes(1);
  })

  test('callback is called when user clicks btnFollow', () => {
    const btnFollow = wrapper.find('#btnFollow');
    // btnFollow.simulate('click');
    btnFollow.prop('onClick')();
    expect(mockProps.onFollowClick).toHaveBeenCalledTimes(1);
  })
});

