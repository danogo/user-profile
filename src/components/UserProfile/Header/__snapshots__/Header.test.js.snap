// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    followers={4433}
    following={723}
    isFollowed={false}
    isLiked={false}
    likes={121}
    onFollowClick={[MockFunction]}
    onHeartClick={[MockFunction]}
    onShareClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btnShare"
          id="btnShare"
          onClick={[MockFunction]}
        >
          <FontAwesomeIcon
            border={false}
            className="shareIcon"
            fixedWidth={false}
            flip={null}
            icon="share-square"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </button>,
        <div
          className="userInfo"
        >
          <img
            alt="user"
            src="pic.jpg"
          />
          <div>
            <h1>
              Harvey Specter
            </h1>
            <button
              className="btnHeart"
              id="btnHeart"
              onClick={[MockFunction]}
            >
              <FontAwesomeIcon
                border={false}
                className="heartIcon"
                color="#D9D9D9"
                fixedWidth={false}
                flip={null}
                icon={
                  Array [
                    "far",
                    "heart",
                  ]
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </button>
            <h2>
              New York, USA
            </h2>
          </div>
        </div>,
        <div
          className="userSocials"
        >
          <ul
            id="ul"
          >
            <li>
              <p
                className="socialsNumber"
              >
                121
              </p>
              <p>
                Likes
              </p>
            </li>
            <span
              className="divider"
            />
            <li>
              <p
                className="socialsNumber"
              >
                723
              </p>
              <p>
                Following
              </p>
            </li>
            <span
              className="divider"
            />
            <li>
              <p
                className="socialsNumber"
              >
                4433
              </p>
              <p>
                Followers
              </p>
            </li>
          </ul>
          <button
            className="btnFollow"
            id="btnFollow"
            onClick={[MockFunction]}
          >
            FOLLOW
          </button>
        </div>,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FontAwesomeIcon
            border={false}
            className="shareIcon"
            fixedWidth={false}
            flip={null}
            icon="share-square"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />,
          "className": "btnShare",
          "id": "btnShare",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "border": false,
            "className": "shareIcon",
            "fixedWidth": false,
            "flip": null,
            "icon": "share-square",
            "inverse": false,
            "listItem": false,
            "mask": null,
            "pull": null,
            "pulse": false,
            "rotation": null,
            "size": null,
            "spin": false,
            "symbol": false,
            "title": "",
            "transform": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="user"
              src="pic.jpg"
            />,
            <div>
              <h1>
                Harvey Specter
              </h1>
              <button
                className="btnHeart"
                id="btnHeart"
                onClick={[MockFunction]}
              >
                <FontAwesomeIcon
                  border={false}
                  className="heartIcon"
                  color="#D9D9D9"
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Array [
                      "far",
                      "heart",
                    ]
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </button>
              <h2>
                New York, USA
              </h2>
            </div>,
          ],
          "className": "userInfo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "user",
              "src": "pic.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Harvey Specter
                </h1>,
                <button
                  className="btnHeart"
                  id="btnHeart"
                  onClick={[MockFunction]}
                >
                  <FontAwesomeIcon
                    border={false}
                    className="heartIcon"
                    color="#D9D9D9"
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Array [
                        "far",
                        "heart",
                      ]
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />
                </button>,
                <h2>
                  New York, USA
                </h2>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Harvey Specter",
                },
                "ref": null,
                "rendered": "Harvey Specter",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <FontAwesomeIcon
                    border={false}
                    className="heartIcon"
                    color="#D9D9D9"
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Array [
                        "far",
                        "heart",
                      ]
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />,
                  "className": "btnHeart",
                  "id": "btnHeart",
                  "onClick": [MockFunction],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "heartIcon",
                    "color": "#D9D9D9",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": Array [
                      "far",
                      "heart",
                    ],
                    "inverse": false,
                    "listItem": false,
                    "mask": null,
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": null,
                    "spin": false,
                    "symbol": false,
                    "title": "",
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New York, USA",
                },
                "ref": null,
                "rendered": "New York, USA",
                "type": "h2",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ul
              id="ul"
            >
              <li>
                <p
                  className="socialsNumber"
                >
                  121
                </p>
                <p>
                  Likes
                </p>
              </li>
              <span
                className="divider"
              />
              <li>
                <p
                  className="socialsNumber"
                >
                  723
                </p>
                <p>
                  Following
                </p>
              </li>
              <span
                className="divider"
              />
              <li>
                <p
                  className="socialsNumber"
                >
                  4433
                </p>
                <p>
                  Followers
                </p>
              </li>
            </ul>,
            <button
              className="btnFollow"
              id="btnFollow"
              onClick={[MockFunction]}
            >
              FOLLOW
            </button>,
          ],
          "className": "userSocials",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <p
                    className="socialsNumber"
                  >
                    121
                  </p>
                  <p>
                    Likes
                  </p>
                </li>,
                <span
                  className="divider"
                />,
                <li>
                  <p
                    className="socialsNumber"
                  >
                    723
                  </p>
                  <p>
                    Following
                  </p>
                </li>,
                <span
                  className="divider"
                />,
                <li>
                  <p
                    className="socialsNumber"
                  >
                    4433
                  </p>
                  <p>
                    Followers
                  </p>
                </li>,
              ],
              "id": "ul",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="socialsNumber"
                    >
                      121
                    </p>,
                    <p>
                      Likes
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 121,
                      "className": "socialsNumber",
                    },
                    "ref": null,
                    "rendered": 121,
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Likes",
                    },
                    "ref": null,
                    "rendered": "Likes",
                    "type": "p",
                  },
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "divider",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="socialsNumber"
                    >
                      723
                    </p>,
                    <p>
                      Following
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 723,
                      "className": "socialsNumber",
                    },
                    "ref": null,
                    "rendered": 723,
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Following",
                    },
                    "ref": null,
                    "rendered": "Following",
                    "type": "p",
                  },
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "divider",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="socialsNumber"
                    >
                      4433
                    </p>,
                    <p>
                      Followers
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 4433,
                      "className": "socialsNumber",
                    },
                    "ref": null,
                    "rendered": 4433,
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Followers",
                    },
                    "ref": null,
                    "rendered": "Followers",
                    "type": "p",
                  },
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "FOLLOW",
              "className": "btnFollow",
              "id": "btnFollow",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "FOLLOW",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btnShare"
            id="btnShare"
            onClick={[MockFunction]}
          >
            <FontAwesomeIcon
              border={false}
              className="shareIcon"
              fixedWidth={false}
              flip={null}
              icon="share-square"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </button>,
          <div
            className="userInfo"
          >
            <img
              alt="user"
              src="pic.jpg"
            />
            <div>
              <h1>
                Harvey Specter
              </h1>
              <button
                className="btnHeart"
                id="btnHeart"
                onClick={[MockFunction]}
              >
                <FontAwesomeIcon
                  border={false}
                  className="heartIcon"
                  color="#D9D9D9"
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Array [
                      "far",
                      "heart",
                    ]
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </button>
              <h2>
                New York, USA
              </h2>
            </div>
          </div>,
          <div
            className="userSocials"
          >
            <ul
              id="ul"
            >
              <li>
                <p
                  className="socialsNumber"
                >
                  121
                </p>
                <p>
                  Likes
                </p>
              </li>
              <span
                className="divider"
              />
              <li>
                <p
                  className="socialsNumber"
                >
                  723
                </p>
                <p>
                  Following
                </p>
              </li>
              <span
                className="divider"
              />
              <li>
                <p
                  className="socialsNumber"
                >
                  4433
                </p>
                <p>
                  Followers
                </p>
              </li>
            </ul>
            <button
              className="btnFollow"
              id="btnFollow"
              onClick={[MockFunction]}
            >
              FOLLOW
            </button>
          </div>,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FontAwesomeIcon
              border={false}
              className="shareIcon"
              fixedWidth={false}
              flip={null}
              icon="share-square"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />,
            "className": "btnShare",
            "id": "btnShare",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "shareIcon",
              "fixedWidth": false,
              "flip": null,
              "icon": "share-square",
              "inverse": false,
              "listItem": false,
              "mask": null,
              "pull": null,
              "pulse": false,
              "rotation": null,
              "size": null,
              "spin": false,
              "symbol": false,
              "title": "",
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="user"
                src="pic.jpg"
              />,
              <div>
                <h1>
                  Harvey Specter
                </h1>
                <button
                  className="btnHeart"
                  id="btnHeart"
                  onClick={[MockFunction]}
                >
                  <FontAwesomeIcon
                    border={false}
                    className="heartIcon"
                    color="#D9D9D9"
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Array [
                        "far",
                        "heart",
                      ]
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />
                </button>
                <h2>
                  New York, USA
                </h2>
              </div>,
            ],
            "className": "userInfo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "user",
                "src": "pic.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Harvey Specter
                  </h1>,
                  <button
                    className="btnHeart"
                    id="btnHeart"
                    onClick={[MockFunction]}
                  >
                    <FontAwesomeIcon
                      border={false}
                      className="heartIcon"
                      color="#D9D9D9"
                      fixedWidth={false}
                      flip={null}
                      icon={
                        Array [
                          "far",
                          "heart",
                        ]
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size={null}
                      spin={false}
                      symbol={false}
                      title=""
                      transform={null}
                    />
                  </button>,
                  <h2>
                    New York, USA
                  </h2>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Harvey Specter",
                  },
                  "ref": null,
                  "rendered": "Harvey Specter",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <FontAwesomeIcon
                      border={false}
                      className="heartIcon"
                      color="#D9D9D9"
                      fixedWidth={false}
                      flip={null}
                      icon={
                        Array [
                          "far",
                          "heart",
                        ]
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size={null}
                      spin={false}
                      symbol={false}
                      title=""
                      transform={null}
                    />,
                    "className": "btnHeart",
                    "id": "btnHeart",
                    "onClick": [MockFunction],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "border": false,
                      "className": "heartIcon",
                      "color": "#D9D9D9",
                      "fixedWidth": false,
                      "flip": null,
                      "icon": Array [
                        "far",
                        "heart",
                      ],
                      "inverse": false,
                      "listItem": false,
                      "mask": null,
                      "pull": null,
                      "pulse": false,
                      "rotation": null,
                      "size": null,
                      "spin": false,
                      "symbol": false,
                      "title": "",
                      "transform": null,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "New York, USA",
                  },
                  "ref": null,
                  "rendered": "New York, USA",
                  "type": "h2",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ul
                id="ul"
              >
                <li>
                  <p
                    className="socialsNumber"
                  >
                    121
                  </p>
                  <p>
                    Likes
                  </p>
                </li>
                <span
                  className="divider"
                />
                <li>
                  <p
                    className="socialsNumber"
                  >
                    723
                  </p>
                  <p>
                    Following
                  </p>
                </li>
                <span
                  className="divider"
                />
                <li>
                  <p
                    className="socialsNumber"
                  >
                    4433
                  </p>
                  <p>
                    Followers
                  </p>
                </li>
              </ul>,
              <button
                className="btnFollow"
                id="btnFollow"
                onClick={[MockFunction]}
              >
                FOLLOW
              </button>,
            ],
            "className": "userSocials",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    <p
                      className="socialsNumber"
                    >
                      121
                    </p>
                    <p>
                      Likes
                    </p>
                  </li>,
                  <span
                    className="divider"
                  />,
                  <li>
                    <p
                      className="socialsNumber"
                    >
                      723
                    </p>
                    <p>
                      Following
                    </p>
                  </li>,
                  <span
                    className="divider"
                  />,
                  <li>
                    <p
                      className="socialsNumber"
                    >
                      4433
                    </p>
                    <p>
                      Followers
                    </p>
                  </li>,
                ],
                "id": "ul",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="socialsNumber"
                      >
                        121
                      </p>,
                      <p>
                        Likes
                      </p>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": 121,
                        "className": "socialsNumber",
                      },
                      "ref": null,
                      "rendered": 121,
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Likes",
                      },
                      "ref": null,
                      "rendered": "Likes",
                      "type": "p",
                    },
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "divider",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="socialsNumber"
                      >
                        723
                      </p>,
                      <p>
                        Following
                      </p>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": 723,
                        "className": "socialsNumber",
                      },
                      "ref": null,
                      "rendered": 723,
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Following",
                      },
                      "ref": null,
                      "rendered": "Following",
                      "type": "p",
                    },
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "divider",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="socialsNumber"
                      >
                        4433
                      </p>,
                      <p>
                        Followers
                      </p>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": 4433,
                        "className": "socialsNumber",
                      },
                      "ref": null,
                      "rendered": 4433,
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Followers",
                      },
                      "ref": null,
                      "rendered": "Followers",
                      "type": "p",
                    },
                  ],
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "FOLLOW",
                "className": "btnFollow",
                "id": "btnFollow",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "FOLLOW",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
